guava/src/com/google/common/hash/AbstractHashFunction.java

```diff
@@ -33,14 +37,17 @@
     return newHasher().putObject(instance, funnel).hash();
   }
 
+  @SuppressWarnings({"lowerbound:argument.type.incompatible",// Since len is length of `input` with min length of 1,
+          // len * 2 can't be negative
```

Comment:
This is true unless `len` is bigger than half of `Integer.MAX_VALUE`. This might be a bug - I don't see anywhere in the documentation of `hashUnencodedChars` that the input's size is limited.

-----

Docs for `HashFunction#hashUnencodedChars`:

```
   * Shortcut for {@code newHasher().putUnencodedChars(input).hash()}. The implementation
   * <i>might</i> perform better than its longhand equivalent, but should not perform worse. Note
   * that no character encoding is performed; the low byte and high byte of each {@code char} are
   * hashed directly (in that order).
   *
   * <p><b>Warning:</b> This method will produce different output than most other languages do when
   * running the same hash function on the equivalent input. For cross-language compatibility, use
   * {@link #hashString}, usually with a charset of UTF-8. For other use cases, use {@code
   * hashUnencodedChars}.
```